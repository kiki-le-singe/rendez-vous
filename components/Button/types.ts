export type TStyles = {
  [key: string]: object;
};

export type themeValues = "green" | "red" | "darkBlue" | "light";

export type ButtonProps = {
  label: string;
  theme?: themeValues;
  onPress?: () => void;
  animatedStyle?: {};
};
