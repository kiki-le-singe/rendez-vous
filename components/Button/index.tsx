import React from "react";
import { StyleSheet, Text, TouchableOpacity } from "react-native";

import { ButtonProps, TStyles } from "./types";
import Colors from "../../constants/Colors";

export default function Button({
  label,
  theme = "light",
  onPress = () => {},
}: ButtonProps) {
  const labelTheme = theme === "light" ? "lightLabel" : "label";

  return (
    <TouchableOpacity
      onPress={onPress}
      style={[styles.container, styles[theme]]}
    >
      <Text style={styles[labelTheme]}>{label}</Text>
    </TouchableOpacity>
  );
}

const styles: TStyles = StyleSheet.create({
  container: {
    padding: 10,
    borderRadius: 6,
  },
  green: {
    backgroundColor: Colors.light.green,
  },
  red: {
    backgroundColor: Colors.light.red,
  },
  darkBlue: {
    backgroundColor: Colors.light.darkBlue,
  },
  light: {
    backgroundColor: Colors.light.white,
    borderWidth: 1,
    borderColor: Colors.light.darkGreen,
  },
  label: {
    color: Colors.light.white,
  },
  lightLabel: {
    color: Colors.light.darkGreen,
  },
});
