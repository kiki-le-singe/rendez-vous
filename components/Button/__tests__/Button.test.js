import * as React from "react";
import renderer from "react-test-renderer";

import Button from "..";

it(`renders correctly`, () => {
  const tree = renderer.create(<Button label="Press me!" />).toJSON();

  expect(tree).toMatchSnapshot();
});
