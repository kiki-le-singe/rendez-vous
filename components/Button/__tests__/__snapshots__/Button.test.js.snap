// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "opacity": 1,
    }
  }
>
  <View
    animatedStyle={
      {
        "value": {},
      }
    }
    collapsable={false}
    style={
      {
        "alignItems": "center",
        "backgroundColor": "#FFFFFF",
        "borderColor": "#34423E",
        "borderRadius": 6,
        "borderWidth": 1,
        "justifyContent": "center",
        "padding": 10,
      }
    }
  >
    <Text
      style={
        {
          "color": "#34423E",
        }
      }
    >
      Press me!
    </Text>
  </View>
</View>
`;
